# Coordonnées géographiques

## 1.Traitement des données

### Importation

In [2]:
import numpy as np
import pandas as pd

# Importation des fonctions dont l'on se servira pour toutes les figures
from bokeh.plotting import figure, output_notebook, show

# Précision de l'affichage des graphiques dans des cellules jupyter
output_notebook()

Loading BokehJS ...

In [3]:
# Définition du chemin d'accès:
### Céline
#%cd C:\Users\c.doussot\Desktop\Data Analyst\GitHub\ParisPyVelib_Datas
### Hermine
%cd C:\Users\h.berthon\Documents\GitHub\ParisPyVelibHB\ParisPyVelib_Datas
### Tarik
#%cd C:\Users\Home\Documents\Git\ParisPyVelib\Data

df = pd.read_csv('2018-2021_comptage-velo-donnees-compteurs.csv')
print(df.nunique())
df.head()


C:\Users\h.berthon\Documents\GitHub\ParisPyVelibHB\ParisPyVelib_Datas
Unnamed: 0          1723879
Id_old                  102
Address1_old            100
Id                       71
Address2_old             76
Count_by_hour          1054
Date_count            29830
Date_instal              37
Photo_old                76
Coord_old                90
Source                    4
Direction                 8
Address                  69
Latitude                 35
Longitude                43
Coord                    50
Coord_full               95
Y_Date_Count              4
M_Date_Count             12
D_Date_Count             31
Dweek_Date_Count          7
H_Date_Count             24
Y_Date_Instal             6
M_Date_Instal             9
D_Date_Instal            21
dtype: int64


Unnamed: 0               Id_old                       Address1_old  \
0           0  100042374-110042374        VOIE GEORGES POMPIDOU NE-SO   
1           1  100042374-109042374        VOIE GEORGES POMPIDOU SO-NE   
2           2         100003098-SC     106 AV DENFERT ROCHEREAU NE-SO   
3           3         100003097-SC             105 RUE LA FAYETTE E-O   
4           4  100036719-103036719  18 QUAI DE L'HOTEL DE VILLE SE-NO   

          Id                    Address2_old  Count_by_hour  \
0  100042374           VOIE GEORGES POMPIDOU            1.0   
1  100042374           VOIE GEORGES POMPIDOU            0.0   
2  100003098  106 AV DENFERT ROCHEREAU NE-SO            3.0   
3  100003097          105 RUE LA FAYETTE E-O            8.0   
4  100036719     18 QUAI DE L'HOTEL DE VILLE            3.0   

            Date_count          Date_instal  \
0  2018-01-01 00:00:00  2017-12-15 00:00:00   
1  2018-01-01 00:00:00  2017-12-15 00:00:00   
2  2018-01-01 00:00:00  2012-02-22 00:00:00   
3  2018-01-01 00:00:00  2012-02-13 00:00:00   
4  2018-01-01 00:00:00  2017-07-12 00:00:00   

                                           Photo_old           Coord_old  ...  \
0  https://www.eco-visio.net/Photos/100042374/151...  48.848399,2.275932  ...   
1  https://www.eco-visio.net/Photos/100042374/151...  48.848399,2.275932  ...   
2  https://www.eco-visio.net/Photos/100003098/133...  48.835143,2.333031  ...   
3  https://www.eco-visio.net/Photos/100003097/157...  48.877667,2.350556  ...   
4  https://www.eco-visio.net/Photos/100036719/149...    48.85372,2.35702  ...   

              Coord              Coord_full Y_Date_Count  M_Date_Count  \
0  48.84800,2.27600  48.84800,2.27600-NE-SO         2018             1   
1  48.84800,2.27600  48.84800,2.27600-SO-NE         2018             1   
2  48.83500,2.33300  48.83500,2.33300-NE-SO         2018             1   
3  48.87800,2.35100    48.87800,2.35100-E-O         2018             1   
4  48.85400,2.35700  48.85400,2.35700-SE-NO         2018             1   

   D_Date_Count Dweek_Date_Count H_Date_Count  Y_Date_Instal  M_Date_Instal  \
0             1                0            0           2017             12   
1             1                0            0           2017             12   
2             1                0            0           2012              2   
3             1                0            0           2012              2   
4             1                0            0           2017              7   

   D_Date_Instal  
0             15  
1             15  
2             22  
3             13  
4             12  

[5 rows x 25 columns]

### Nouveau DataFrame

In [4]:
#Nouveau df avec moins de colonnes
df_lat_lon= df.drop(['Unnamed: 0','Id_old','Address1_old', 'Address2_old','Date_count', 'Date_instal', 'Photo_old', 'Coord_full', 
                      'Latitude', 'Longitude', 'Direction', 'Y_Date_Count', 'M_Date_Count', 'D_Date_Count',
                     'Dweek_Date_Count', 'H_Date_Count', 'Y_Date_Instal', 'M_Date_Instal', 'D_Date_Instal', 'Coord_full'
                     ], axis =1)
df_lat_lon= df_lat_lon.rename({'Coord': 'Coord_simpl'}, axis = 1)
df_lat_lon.head()

Id  Count_by_hour           Coord_old  Source  \
0  100042374            1.0  48.848399,2.275932    2018   
1  100042374            0.0  48.848399,2.275932    2018   
2  100003098            3.0  48.835143,2.333031    2018   
3  100003097            8.0  48.877667,2.350556    2018   
4  100036719            3.0    48.85372,2.35702    2018   

                       Address       Coord_simpl  
0        VOIE GEORGES POMPIDOU  48.84800,2.27600  
1        VOIE GEORGES POMPIDOU  48.84800,2.27600  
2     106 AV DENFERT ROCHEREAU  48.83500,2.33300  
3           105 RUE LA FAYETTE  48.87800,2.35100  
4  18 QUAI DE L'HOTEL DE VILLE  48.85400,2.35700

In [5]:
#Changement de format de coordonnées géo
# Création des colonnes 'longitude', et 'latitude'
df_lat_lon[['Latitude', 'Longitude']] = df_lat_lon['Coord_old'].str.split(',', expand = True)

#Les coordonnées que nous avons sont au format Degrés décimaux (DD) = 5 décimales
df_lat_lon['Latitude'] = df_lat_lon['Latitude'].astype('float64').round(5)
df_lat_lon['Longitude'] = df_lat_lon['Longitude'].astype('float64').round(5)

#Nouvelle colonne avec les coordonnées arrondies
df_lat_lon['Coord'] = df_lat_lon['Latitude'].astype('str') + ',' + df_lat_lon['Longitude'].astype('str')

#Suppression colonnes inutiles
df_lat_lon = df_lat_lon.drop(['Coord_old', 'Latitude', 'Longitude'], axis=1)

#Tri par adresse
df_lat_lon = df_lat_lon.sort_values(by = ['Address'], axis=0)

df_lat_lon.head()

Id  Count_by_hour  Source                   Address  \
1395687  100044494            8.0    2021  10 AV DE LA GRANDE ARMÉE   
749911   100044494            6.0    2020  10 AV DE LA GRANDE ARMÉE   
152347   100044494            0.0    2018  10 AV DE LA GRANDE ARMÉE   
326371   100044494            1.0    2019  10 AV DE LA GRANDE ARMÉE   
1026401  100044494            4.0    2020  10 AV DE LA GRANDE ARMÉE   

              Coord_simpl             Coord  
1395687  48.87500,2.29200  48.87472,2.29244  
749911   48.87500,2.29200  48.87472,2.29244  
152347   48.87500,2.29200  48.87472,2.29244  
326371   48.87500,2.29200  48.87472,2.29244  
1026401  48.87500,2.29200  48.87472,2.29244

In [6]:
#Nombre de valeurs uniques avant le traitement des données
df_lat_lon.nunique()

Id                 71
Count_by_hour    1054
Source              4
Address            69
Coord_simpl        50
Coord              84
dtype: int64

Pour pouvoir représenter les différentes addresses sur une carte intéractive, il faut que les addresses correspondent aux coordonnées

### Traitement des addresses et coordonnées géographiques

Je n'ai pas réussi à faire de programme pour remplacer les données, j'ai donc fait le tout manuellement

In [7]:
df_lat_lon[['Id','Coord','Address', 'Source']].groupby(['Address', 'Id','Coord', 'Source']).count().head(15)

Empty DataFrame
Columns: []
Index: [(10 AV DE LA GRANDE ARMÉE, 100044494, 48.87472,2.29244, 2018), (10 AV DE LA GRANDE ARMÉE, 100044494, 48.87472,2.29244, 2019), (10 AV DE LA GRANDE ARMÉE, 100044494, 48.87472,2.29244, 2020), (10 AV DE LA GRANDE ARMÉE, 100044494, 48.87472,2.29244, 2021), (10 BD AUGUSTE BLANQUI, 100049408, 48.83068,2.35348, 2020), (10 BD AUGUSTE BLANQUI, 100049408, 48.83068,2.35348, 2021), (10 BD AUGUSTE BLANQUI, 100049408, 48.8309,2.35324, 2018), (10 BD AUGUSTE BLANQUI, 100049408, 48.8309,2.35324, 2019), (100 RUE LA FAYETTE, 100003099, 48.87746,2.35008, 2018), (100 RUE LA FAYETTE, 100003099, 48.87746,2.35008, 2019), (100 RUE LA FAYETTE, 100003099, 48.87746,2.35008, 2020), (100 RUE LA FAYETTE, 100003099, 48.87746,2.35008, 2021), (102 BD DE MAGENTA, 100047536, 48.87773,2.35493, 2020), (102 BD DE MAGENTA, 100047536, 48.87773,2.35493, 2021), (102 BD MAGENTA, 100047536, 48.87773,2.35493, 2018)]

On remarque que la notation de certaines addresses a évolué en 2020, de même pour les coordonnées géographiques, pour simplifier je vais modifier les addresses et coordonnées géographiques en prenant la valeur récente de 2021

In [8]:
#Groupby pour voir de manière globale les différences entre addresses et coordonnées géographiques
df_lat_lon[['Address','Id','Coord',]].groupby(['Coord','Id','Address']).count().head(60)

Empty DataFrame
Columns: []
Index: [(48.82024,2.35902, 100056039, 180 AV D'ITALIE), (48.82026,2.3592, 100056041, 147 AV D'ITALIE), (48.82108,2.32537, 100056225, 3 AV DE LA PORTE D'ORLÉANS), (48.82203,2.32545, 100056225, 3 AV DE LA PORTE D'ORLÉANS), (48.82636,2.30303, 100047547, 6 RUE JULIA BARTET), (48.82648,2.30315, 100047547, 6 RUE JULIA BARTET), (48.8265,2.38434, 100047541, PONT NATIONAL), (48.82658,2.38409, 100047541, PONT NATIONAL), (48.82682,2.38465, 100047549, PONT NATIONAL), (48.82952,2.38699, 100047546, FACE AU 70 QUAI DE BERCY), (48.83033,2.40055, 100056226, FACE AU 8 AV DE LA PORTE DE CHARENTON), (48.83045,2.3532, 100047539, 21 BD AUGUSTE BLANQUI), (48.83068,2.35348, 100049408, 10 BD AUGUSTE BLANQUI), (48.8309,2.35324, 100049408, 10 BD AUGUSTE BLANQUI), (48.83421,2.26542, 100056331, FACE AU 40 QUAI D'ISSY), (48.83436,2.377, 100036718, 39 QUAI FRANÇOIS MAURIAC), (48.8344,2.37694, 100036718, 39 QUAI FRANÇOIS MAURIAC), (48.8347,2.33297, 100003096, 97 AV DENFERT ROCHEREAU), (48.83511,2.33338, 100003096, 97 AV DENFERT ROCHEREAU), (48.83514,2.33303, 100003098, 106 AV DENFERT ROCHEREAU), (48.83521,2.33307, 100003098, 106 AV DENFERT ROCHEREAU), (48.83848,2.37587, 100056332, PONT DE BERCY), (48.83977,2.30198, 300014702, 254 RUE DE VAUGIRARD), (48.83988,2.2671, 100047551, PONT DU GARIGLIANO), (48.83992,2.26694, 100047551, PONT DU GARIGLIANO), (48.83993,2.26715, 100047535, PONT DU GARIGLIANO), (48.84015,2.26733, 100047535, PONT DU GARIGLIANO), (48.8408,2.33323, 100049407, 152 BD DU MONTPARNASSE), (48.84201,2.36729, 100057329, TOTEM 85 QUAI D'AUSTERLITZ), (48.84209,2.301, 100056040, 129 RUE LECOURBE), (48.84216,2.30115, 100056044, 132 RUE LECOURBE), (48.84223,2.36811, 100056329, PONT CHARLES DE GAULLE), (48.84344,2.38338, 100006300, 135 AV DAUMESNIL), (48.84603,2.37543, 100007049, 28 BD DIDEROT), (48.8461,2.37546, 100041488, 27 BD DIDEROT), (48.84638,2.31529, 100060178, 90 RUE DE SÈVRES), (48.8484,2.27586, 100042374, VOIE GEORGES POMPIDOU), (48.8484,2.27593, 100042374, VOIE GEORGES POMPIDOU), (48.85013,2.35423, 100056336, 27 QUAI DE LA TOURNELLE), (48.85113,2.34568, 100060175, 30 RUE SAINT JACQUES), (48.85152,2.3433, 100060174, 21 BD SAINT MICHEL), (48.85204,2.28589, 100056330, 36 QUAI DE GRENELLE), (48.85209,2.28508, 100056330, 36 QUAI DE GRENELLE), (48.85372,2.35702, 100036719, 18 QUAI DE L'HOTEL DE VILLE), (48.85735,2.35211, 100050876, TOTEM 64 RUE DE RIVOLI), (48.86053,2.38836, 100047550, 35 BD DE MENILMONTANT), (48.86053,2.38836, 100065336, 35 BD DE MÉNILMONTANT), (48.86057,2.38886, 100047537, 26 BD DE MÉNILMONTANT), (48.86062,2.38872, 100047537, 26 BD DE MÉNILMONTANT), (48.86077,2.37274, 100047540, 72 BD RICHARD LENOIR), (48.86077,2.37305, 100047540, 72 BD RICHARD LENOIR), (48.86085,2.37228, 100047544, 77 BD RICHARD LENOIR), (48.86149,2.37376, 100044493, 67 BD VOLTAIRE), (48.86155,2.37407, 100044506, 72 BD VOLTAIRE), (48.86179,2.32014, 100050132, 243 BD SAINT GERMAIN), (48.86282,2.31061, 100056046, PONT DES INVALIDES), (48.86284,2.31034, 100056223, PONT DES INVALIDES), (48.86288,2.31179, 100056335, QUAI D'ORSAY), (48.86377,2.35096, 100057445, TOTEM 73 BD DE SÉBASTOPOL), (48.86378,2.32003, 100056047, PONT DE LA CONCORDE)]

In [9]:
#Vérification des cohérences entre adresse, Id et coordonnées géographiques
df_inter = df_lat_lon[(df_lat_lon['Id'] == 100065336)| (df_lat_lon['Id'] == 100047550)]
df_inter[['Id','Coord','Address', 'Source']].groupby([ 'Id','Address','Coord', 'Source']).count()

Empty DataFrame
Columns: []
Index: [(100047550, 35 BD DE MENILMONTANT, 48.86053,2.38836, 2018), (100047550, 35 BD DE MENILMONTANT, 48.86053,2.38836, 2019), (100065336, 35 BD DE MÉNILMONTANT, 48.86053,2.38836, 2021)]

En utilisant le groupby on peut identifier de manière plus précise les Id posant problème et choisir la donnée la plus récente

In [10]:
#Certaines adresses présentent des espaces qui sont gommées à l'affichage du groupby
for i in df_lat_lon['Address'].unique():
    print(i)

10 AV DE LA GRANDE ARMÉE
10 BD AUGUSTE BLANQUI
100 RUE LA FAYETTE
102 BD DE MAGENTA
102 BD MAGENTA
105 RUE LA FAYETTE
106 AV DENFERT ROCHEREAU
129 RUE LECOURBE
132 RUE LECOURBE
135 AV DAUMESNIL
147 AV D'ITALIE
152 BD DU MONTPARNASSE
16 AV DE LA PORTE DES TERNES
18 QUAI DE L'HOTEL DE VILLE
180 AV D'ITALIE
2 AV DE LA PORTE DE BAGNOLET
20 AV DE CLICHY
21 BD AUGUSTE BLANQUI
21 BD SAINT MICHEL
243 BD SAINT GERMAIN
254 RUE DE VAUGIRARD
26 BD DE MÉNILMONTANT
27 BD DIDEROT
27 QUAI DE LA TOURNELLE
28 BD DIDEROT
3 AV DE LA PORTE D'ORLÉANS
30 RUE SAINT JACQUES
33 AV DES CHAMPS ELYSÉES
35 BD DE MENILMONTANT
35 BD DE MÉNILMONTANT
36 QUAI DE GRENELLE
38 RUE TURBIGO
39 QUAI FRANÇOIS MAURIAC
44 AV DES CHAMPS ELYSÉES
6 RUE JULIA BARTET
67 BD VOLTAIRE
69 BD ORNANO
7 AV DE LA GRANDE ARMÉE
72 AV DE FLANDRE
72 AV FLANDRE
72 BD RICHARD LENOIR
72 BD VOLTAIRE
72 RUE DE LA CHAPELLE
74 BD ORNANO
77 BD RICHARD LENOIR
87 AV DE FLANDRE
89 BD DE MAGENTA
90 RUE DE SÈVRES
97 AV DENFERT ROCHEREAU
FACE 104 RUE D'AUBERV

#Remplacement au niveau des adresses
df_lat_lon['Address'] = df_lat_lon['Address'].replace({'10  AVENUE DE LA GRANDE ARMÉE SE-NO' : '10 AVENUE DE LA GRANDE ARMÉE',
                                                      '10 AVENUE DE LA GRANDE ARMÉE SE-NO' : '10 AVENUE DE LA GRANDE ARMÉE',
                                                      '100 RUE LA FAYETTE O-E' : '100 RUE LA FAYETTE',
                                                      '102 BOULEVARD DE MAGENTA SE-NO':'102 BD MAGENTA SE-NO',
                                                      '10 BOULEVARD AUGUSTE BLANQUI NE-SO' : '10 BD AUGUSTE BLANQUI NE-SO',
                                                      '72 AVENUE FLANDRE SO-NE':'72 AVENUE DE FLANDRE SO-NE',
                                                      '97 AVENUE DENFERT ROCHEREAU SO-NE':'97 AVENUE DENFERT ROCHEREAU',
                                                      "18 QUAI DE L'HOTEL DE VILLE": "18 QUAI DE L'HÔTEL DE VILLE",
                                                      'PONT DU GARIGLIANO SE-NO':'PONT DU GARIGLIANO',
                                                       '77 BOULEVARD RICHARD LENOIR N-S':'77 BD RICHARD LENOIR N-S',
                                                       '35 BOULEVARD DE MENILMONTANT NO-SE':'35 BOULEVARD DE MÉNILMONTANT',
                                                       '72 BOULEVARD RICHARD LENOIR  S-N':'72 BD RICHARD LENOIR  S-N'
                                                       
                                                      })

In [11]:
#Remplacement au niveau des Lat & Long
df_lat_lon['Coord'] = df_lat_lon['Coord'].replace({'48.8309,2.35324' : '48.83068,2.35348',
                                                   '48.83514,2.33303' : '48.83521,2.33307',
                                                  '48.86062,2.38872' : '48.86057,2.38886',
                                                  '48.82203,2.32545' : '48.82108,2.32537',
                                                  '48.85204,2.28589' : '48.85209,2.28508',
                                                  '48.8344,2.37694':'48.83436,2.377',
                                                  '48.82648,2.30315': '48.82636,2.30303',
                                                  '48.8484,2.27593':'48.8484,2.27586',
                                                 '48.8347,2.33297':'48.83511,2.33338',
                                                  '48.83993,2.26715':'48.84015,2.26733',
                                                   '48.8265,2.38434':'48.82658,2.38409',
                                                   '48.83988,2.2671':'48.83992,2.26694',
                                                   '48.86077,2.37274':'48.86077,2.37305',
                                                   '48.89141,2.38495':'48.89142,2.38495'
                                                  })
#Remplacement identifiant
df_lat_lon['Id'] = df_lat_lon['Id'].replace({100047550:100065336})

In [12]:
df_lat_lon.nunique()

Id                 70
Count_by_hour    1054
Source              4
Address            69
Coord_simpl        50
Coord              70
dtype: int64

Les identifiant, addresses et coordonnées géographiques sont les mêmes pour toutes les données provenant de 2018 à 2021

On peut maintenant séparer les coordonnées géographiques

In [13]:
df_lat_lon[['Latitude', 'Longitude']] = df_lat_lon['Coord'].str.split(',', expand = True)

#Suppression colonnes inutiles
df_lat_lon = df_lat_lon.drop(['Coord'], axis=1)

df_lat_lon.head()

Id  Count_by_hour  Source                   Address  \
1395687  100044494            8.0    2021  10 AV DE LA GRANDE ARMÉE   
749911   100044494            6.0    2020  10 AV DE LA GRANDE ARMÉE   
152347   100044494            0.0    2018  10 AV DE LA GRANDE ARMÉE   
326371   100044494            1.0    2019  10 AV DE LA GRANDE ARMÉE   
1026401  100044494            4.0    2020  10 AV DE LA GRANDE ARMÉE   

              Coord_simpl  Latitude Longitude  
1395687  48.87500,2.29200  48.87472   2.29244  
749911   48.87500,2.29200  48.87472   2.29244  
152347   48.87500,2.29200  48.87472   2.29244  
326371   48.87500,2.29200  48.87472   2.29244  
1026401  48.87500,2.29200  48.87472   2.29244

## 2. Mapping

In [14]:
#Il faut changer coord en mercator
import math

# derived from the Java version explained here: http://wiki.openstreetmap.org/wiki/Mercator
RADIUS = 6378137.0 # in meters on the equator

def lat2y(a):
    return math.log(math.tan(math.pi / 4 + math.radians(a) / 2)) * RADIUS

def lon2x(a):
    return math.radians(a) * RADIUS

In [15]:
df_lat_lon.info()
#Changement de type de Latitude et Longitude

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1723879 entries, 1395687 to 0
Data columns (total 7 columns):
 #   Column         Dtype  
---  ------         -----  
 0   Id             int64  
 1   Count_by_hour  float64
 2   Source         int64  
 3   Address        object 
 4   Coord_simpl    object 
 5   Latitude       object 
 6   Longitude      object 
dtypes: float64(1), int64(2), object(4)
memory usage: 105.2+ MB


In [16]:
df_lat_lon['Latitude'] = df_lat_lon['Latitude'].astype('float64')
df_lat_lon['Longitude'] = df_lat_lon['Longitude'].astype('float64')

In [17]:
#Vérification
df_lat_lon.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1723879 entries, 1395687 to 0
Data columns (total 7 columns):
 #   Column         Dtype  
---  ------         -----  
 0   Id             int64  
 1   Count_by_hour  float64
 2   Source         int64  
 3   Address        object 
 4   Coord_simpl    object 
 5   Latitude       float64
 6   Longitude      float64
dtypes: float64(3), int64(2), object(2)
memory usage: 105.2+ MB


In [18]:
df_lat_lon['Latitude M'] = df_lat_lon['Latitude'].apply(lat2y)
df_lat_lon['Longitude M'] = df_lat_lon['Longitude'].apply(lon2x)

In [19]:
df_lat_lon.head()

Id  Count_by_hour  Source                   Address  \
1395687  100044494            8.0    2021  10 AV DE LA GRANDE ARMÉE   
749911   100044494            6.0    2020  10 AV DE LA GRANDE ARMÉE   
152347   100044494            0.0    2018  10 AV DE LA GRANDE ARMÉE   
326371   100044494            1.0    2019  10 AV DE LA GRANDE ARMÉE   
1026401  100044494            4.0    2020  10 AV DE LA GRANDE ARMÉE   

              Coord_simpl  Latitude  Longitude    Latitude M    Longitude M  
1395687  48.87500,2.29200  48.87472    2.29244  6.253631e+06  255193.253474  
749911   48.87500,2.29200  48.87472    2.29244  6.253631e+06  255193.253474  
152347   48.87500,2.29200  48.87472    2.29244  6.253631e+06  255193.253474  
326371   48.87500,2.29200  48.87472    2.29244  6.253631e+06  255193.253474  
1026401  48.87500,2.29200  48.87472    2.29244  6.253631e+06  255193.253474

Pour une meilleure lecture avec Bokeh, il faut supprimer les apostrophes " ' "

In [20]:
df_lat_lon['Address'] = df_lat_lon['Address'].str.replace(r"[\"\',]", '')

Définition des bornes

In [21]:
print('Latitude min : ', df_lat_lon['Latitude M'].min())
print('Latitude max : ', df_lat_lon['Latitude M'].max())
print('Longitude min : ', df_lat_lon['Longitude M'].min())
print('Longitude max : ', df_lat_lon['Longitude M'].max())

Latitude min :  6244414.743721515
Latitude max :  6257395.712802779
Longitude min :  252185.40083289784
Longitude max :  268245.4637696434


In [22]:
#On groupe le dataFrame pour avoir le nombre de comptage total par adresse
df2 = df_lat_lon.groupby(['Address','Latitude', 'Longitude', 'Latitude M', 'Longitude M']).agg({'Count_by_hour' : lambda Count_by_hour : Count_by_hour.count()}).reset_index()
df2 = df2.sort_values(by = ['Count_by_hour'],ascending=False, axis=0)
df2.head(60)

Address  Latitude  Longitude    Latitude M  \
13             18 QUAI DE LHOTEL DE VILLE  48.85372    2.35702  6.250077e+06   
32               39 QUAI FRANÇOIS MAURIAC  48.83436    2.37700  6.246802e+06   
24                          28 BD DIDEROT  48.84603    2.37543  6.248776e+06   
72                  VOIE GEORGES POMPIDOU  48.84840    2.27586  6.249177e+06   
51               FACE AU 25 QUAI DE LOISE  48.89142    2.38495  6.256458e+06   
34                     6 RUE JULIA BARTET  48.82636    2.30303  6.245450e+06   
54            FACE AU 48 QUAI DE LA MARNE  48.89122    2.38573  6.256424e+06   
49            FACE 104 RUE DAUBERVILLIERS  48.89046    2.36885  6.256295e+06   
55               FACE AU 70 QUAI DE BERCY  48.82952    2.38699  6.245984e+06   
11                 152 BD DU MONTPARNASSE  48.84080    2.33323  6.247892e+06   
68                 TOTEM 64 RUE DE RIVOLI  48.85735    2.35211  6.250691e+06   
6                106 AV DENFERT ROCHEREAU  48.83521    2.33307  6.246946e+06   
48                97 AV DENFERT ROCHEREAU  48.83511    2.33338  6.246929e+06   
5                      105 RUE LA FAYETTE  48.87767    2.35056  6.254130e+06   
22                          27 BD DIDEROT  48.84610    2.37546  6.248788e+06   
2                      100 RUE LA FAYETTE  48.87746    2.35008  6.254094e+06   
9                        135 AV DAUMESNIL  48.84344    2.38338  6.248338e+06   
56  FACE AU 8 AV DE LA PORTE DE CHARENTON  48.83033    2.40055  6.246121e+06   
42                  72 RUE DE LA CHAPELLE  48.89594    2.35953  6.257223e+06   
53                  FACE AU 40 QUAI DISSY  48.83421    2.26542  6.246777e+06   
52   FACE AU 4 AV DE LA PORTE DE BAGNOLET  48.86461    2.40969  6.251920e+06   
67                            QUAI DORSAY  48.86288    2.31179  6.251627e+06   
30                    36 QUAI DE GRENELLE  48.85209    2.28508  6.249801e+06   
23                27 QUAI DE LA TOURNELLE  48.85013    2.35423  6.249470e+06   
31                         38 RUE TURBIGO  48.86521    2.35358  6.252021e+06   
58                          PONT DE BERCY  48.83848    2.37587  6.247499e+06   
57                 PONT CHARLES DE GAULLE  48.84223    2.36811  6.248133e+06   
41                         72 BD VOLTAIRE  48.86155    2.37407  6.251402e+06   
35                         67 BD VOLTAIRE  48.86149    2.37376  6.251392e+06   
37                7 AV DE LA GRANDE ARMÉE  48.87451    2.29215  6.253595e+06   
0                10 AV DE LA GRANDE ARMÉE  48.87472    2.29244  6.253631e+06   
69              TOTEM 73 BD DE SÉBASTOPOL  48.86377    2.35096  6.251778e+06   
71                   TOTEM COURS LA REINE  48.86462    2.31444  6.251921e+06   
70              TOTEM 85 QUAI DAUSTERLITZ  48.84201    2.36729  6.248096e+06   
64                     PONT DU GARIGLIANO  48.84015    2.26733  6.247782e+06   
46                       89 BD DE MAGENTA  48.87769    2.35447  6.254133e+06   
12           16 AV DE LA PORTE DES TERNES  48.88181    2.28155  6.254831e+06   
21                  26 BD DE MÉNILMONTANT  48.86057    2.38886  6.251236e+06   
65                          PONT NATIONAL  48.82658    2.38409  6.245487e+06   
40                   72 BD RICHARD LENOIR  48.86077    2.37305  6.251270e+06   
17                  21 BD AUGUSTE BLANQUI  48.83045    2.35320  6.246141e+06   
44                   77 BD RICHARD LENOIR  48.86085    2.37228  6.251283e+06   
66                          PONT NATIONAL  48.82682    2.38465  6.245527e+06   
1                   10 BD AUGUSTE BLANQUI  48.83068    2.35348  6.246180e+06   
63                     PONT DU GARIGLIANO  48.83992    2.26694  6.247743e+06   
19                   243 BD SAINT GERMAIN  48.86179    2.32014  6.251443e+06   
50   FACE AU 16 AV DE LA PORTE DES TERNES  48.88163    2.28120  6.254800e+06   
16                        20 AV DE CLICHY  48.88529    2.32666  6.255420e+06   
47                       90 RUE DE SÈVRES  48.84638    2.31529  6.248835e+06   
7                        129 RUE L

In [23]:
df2.nunique()

Address          69
Latitude         70
Longitude        70
Latitude M       70
Longitude M      70
Count_by_hour    51
dtype: int64

In [24]:
#Top 10 des addresses avec le plus passage total
#2018
df_2018 = df_lat_lon[(df_lat_lon['Source'] == 2018)]
df2018 = df_2018.groupby(['Address','Latitude', 'Longitude', 'Latitude M', 'Longitude M', 'Source']).agg({'Count_by_hour' : lambda Count_by_hour : Count_by_hour.count()}).reset_index()
df2018 = df2018.sort_values(by = ['Count_by_hour'],ascending=False, axis=0)
df2018 = df2018.head(10)


#2019
df_2019 = df_lat_lon[df_lat_lon.Source == 2019]
df2019 = df_2019.groupby(['Address','Latitude', 'Longitude', 'Latitude M', 'Longitude M', 'Source']).agg({'Count_by_hour' : lambda Count_by_hour : Count_by_hour.count()}).reset_index()
df2019 = df2019.sort_values(by = ['Count_by_hour'],ascending=False, axis=0)
df2019 = df2019.head(10)

#2020
df_2020 = df_lat_lon[df_lat_lon.Source == 2020]
df2020 = df_2020.groupby(['Address','Latitude', 'Longitude', 'Latitude M', 'Longitude M', 'Source']).agg({'Count_by_hour' : lambda Count_by_hour : Count_by_hour.count()}).reset_index()
df2020 = df2020.sort_values(by = ['Count_by_hour'],ascending=False, axis=0)
df2020 = df2020.head(10)

#2021
df_2021 = df_lat_lon[df_lat_lon.Source == 2021]
df2021 = df_2021.groupby(['Address','Latitude', 'Longitude', 'Latitude M', 'Longitude M', 'Source']).agg({'Count_by_hour' : lambda Count_by_hour : Count_by_hour.count()}).reset_index()
df2021 = df2021.sort_values(by = ['Count_by_hour'],ascending=False, axis=0)
df2021 = df2021.head(10)

In [25]:
from bokeh.tile_providers import get_provider
from bokeh.models import ColumnDataSource, LabelSet
from bokeh.models.tools import HoverTool
from bokeh.models.widgets import Panel, Tabs

source = ColumnDataSource(df2)
source2018 = ColumnDataSource(df2018)
source2019 = ColumnDataSource(df2019)
source2020 = ColumnDataSource(df2020)
source2021 = ColumnDataSource(df2021)

hover = HoverTool(tooltips=[("Address", "@Address"), ("Lat", "@Latitude"), ("Lng", "@Longitude")])

p = figure(x_range = (250000, 270000), y_range = (6244414, 6257395), 
           x_axis_type = 'mercator', y_axis_type = 'mercator')
p2018 = figure(x_range = (250000, 270000), y_range = (6244414, 6257395), 
           x_axis_type = 'mercator', y_axis_type = 'mercator')
p2019 = figure(x_range = (250000, 270000), y_range = (6244414, 6257395), 
           x_axis_type = 'mercator', y_axis_type = 'mercator')
p2020 = figure(x_range = (250000, 270000), y_range = (6244414, 6257395), 
           x_axis_type = 'mercator', y_axis_type = 'mercator')
p2021 = figure(x_range = (250000, 270000), y_range = (6244414, 6257395), 
           x_axis_type = 'mercator', y_axis_type = 'mercator')

tuile = get_provider('OSM')
p.add_tile(tuile)
p2018.add_tile(tuile)
p2019.add_tile(tuile)
p2020.add_tile(tuile)
p2021.add_tile(tuile)

p.circle(x='Longitude M', y='Latitude M',source = source, fill_color = 'blue',line_color='black', size = 10)
p2018.circle(x='Longitude M', y='Latitude M',source = source2018, fill_color = 'red',line_color='black', size = 10)
p2019.circle(x='Longitude M', y='Latitude M',source = source2019, fill_color = 'purple',line_color='black', size = 10)
p2020.circle(x='Longitude M', y='Latitude M',source = source2020, fill_color = 'deeppink',line_color='black', size = 10)
p2021.circle(x='Longitude M', y='Latitude M',source = source2021, fill_color = 'coral',line_color='black', size = 10)

p.add_tools(hover)
p2018.add_tools(hover)
p2019.add_tools(hover)
p2020.add_tools(hover)
p2021.add_tools(hover)

tab = Panel(child = p, title = 'Carte de Paris')
tab2018 = Panel(child = p2018, title = 'top de 2018')
tab2019 = Panel(child = p2019, title = 'top de 2019')
tab2020 = Panel(child = p2020, title = 'top de 2020')
tab2021 = Panel(child = p2021, title = 'top de 2021')

tabs= Tabs(tabs=[tab, tab2018, tab2019, tab2020, tab2021])
show(tabs)

In [40]:
df2018['Color']= 'deeppink'
df2019['Color']= 'coral'
df2020['Color']= 'red'
df2021['Color']= 'purple'
df_top = pd.concat([df2018, df2019, df2020, df2021])
df_top

Address  Latitude  Longitude    Latitude M  \
9              18 QUAI DE LHOTEL DE VILLE  48.85372    2.35702  6.250077e+06   
14                          28 BD DIDEROT  48.84603    2.37543  6.248776e+06   
34                  VOIE GEORGES POMPIDOU  48.84840    2.27586  6.249177e+06   
16               39 QUAI FRANÇOIS MAURIAC  48.83436    2.37700  6.246802e+06   
13                          27 BD DIDEROT  48.84610    2.37546  6.248788e+06   
2                      100 RUE LA FAYETTE  48.87746    2.35008  6.254094e+06   
24                97 AV DENFERT ROCHEREAU  48.83511    2.33338  6.246929e+06   
4                      105 RUE LA FAYETTE  48.87767    2.35056  6.254130e+06   
5                106 AV DENFERT ROCHEREAU  48.83521    2.33307  6.246946e+06   
6                        135 AV DAUMESNIL  48.84344    2.38338  6.248338e+06   
59                  VOIE GEORGES POMPIDOU  48.84840    2.27586  6.249177e+06   
44            FACE AU 48 QUAI DE LA MARNE  48.89122    2.38573  6.256424e+06   
26               39 QUAI FRANÇOIS MAURIAC  48.83436    2.37700  6.246802e+06   
12             18 QUAI DE LHOTEL DE VILLE  48.85372    2.35702  6.250077e+06   
41               FACE AU 25 QUAI DE LOISE  48.89142    2.38495  6.256458e+06   
10                 152 BD DU MONTPARNASSE  48.84080    2.33323  6.247892e+06   
45               FACE AU 70 QUAI DE BERCY  48.82952    2.38699  6.245984e+06   
39            FACE 104 RUE DAUBERVILLIERS  48.89046    2.36885  6.256295e+06   
28                     6 RUE JULIA BARTET  48.82636    2.30303  6.245450e+06   
20                          28 BD DIDEROT  48.84603    2.37543  6.248776e+06   
68                  VOIE GEORGES POMPIDOU  48.84840    2.27586  6.249177e+06   
54                          PONT DE BERCY  48.83848    2.37587  6.247499e+06   
31                     6 RUE JULIA BARTET  48.82636    2.30303  6.245450e+06   
29               39 QUAI FRANÇOIS MAURIAC  48.83436    2.37700  6.246802e+06   
28                         38 RUE TURBIGO  48.86521    2.35358  6.252021e+06   
27                    36 QUAI DE GRENELLE  48.85209    2.28508  6.249801e+06   
45            FACE 104 RUE DAUBERVILLIERS  48.89046    2.36885  6.256295e+06   
47               FACE AU 25 QUAI DE LOISE  48.89142    2.38495  6.256458e+06   
48   FACE AU 4 AV DE LA PORTE DE BAGNOLET  48.86461    2.40969  6.251920e+06   
23                          28 BD DIDEROT  48.84603    2.37543  6.248776e+06   
54                 PONT CHARLES DE GAULLE  48.84223    2.36811  6.248133e+06   
49   FACE AU 4 AV DE LA PORTE DE BAGNOLET  48.86461    2.40969  6.251920e+06   
53  FACE AU 8 AV DE LA PORTE DE CHARENTON  48.83033    2.40055  6.246121e+06   
22                27 QUAI DE LA TOURNELLE  48.85013    2.35423  6.249470e+06   
23                          28 BD DIDEROT  48.84603    2.37543  6.248776e+06   
52               FACE AU 70 QUAI DE BERCY  48.82952    2.38699  6.245984e+06   
51            FACE AU 48 QUAI DE LA MARNE  48.89122    2.38573  6.256424e+06   
50                  FACE AU 40 QUAI DISSY  48.83421    2.26542  6.246777e+06   
28                    36 QUAI DE GRENELLE  48.85209    2.28508  6.249801e+06   
55                          PONT DE BERCY  48.83848    2.37587  6.247499e+06   

      Longitude M  Source  Count_by_hour     Color  
9   262382.266190    2018        17474.0  deeppink  
14  264431.658015    2018        17474.0  deeppink  
34  253347.576317    2018        17473.0  deeppink  
16  264606.429616    2018        17473.0  deeppink  
13  264434.997600    2018         8737.0  deeppink  
2   261609.708923    2018         8737.0  deeppink  
24  259750.673427    2018         8737.0  deeppink  
4   261663.142279    2018         8737.0  deeppink  
5   259716.164385    2018         8737.0  deeppink  
6   265316.647967    2018         8737.0  deeppink  
59  253347.576317    2019        17474.0     coral  
44  265578.248770    2019        17474.0     coral  
26  264606.429616    2019        17474.0     coral  
12  262382.

In [41]:
source = ColumnDataSource(df_top)


hover = HoverTool(tooltips=[("Address", "@Address"), ("Lat", "@Latitude"), ("Lng", "@Longitude")])

p = figure(x_range = (250000, 270000), y_range = (6244414, 6257395), 
           x_axis_type = 'mercator', y_axis_type = 'mercator')


tuile = get_provider('OSM')
p.add_tile(tuile)

p.circle(x='Longitude M', y='Latitude M',source = source, fill_color = 'Color',line_color='black', size = 10)

p.add_tools(hover)

show(p)

En représentant les différents top selon les années, il est posssible de voir une tendance avec plus de circulation autour de la Seine, près des axes ferroviers et le long des boulevards importants.
Il est également intéressant de voir que les top évoluent, certainement grâce à l'installation de nouveaux compteurs.